eras/shelley-ma/test-suite/test/Test/Cardano/Ledger/Mary/Value.hs

```diff
@@ -354,6 +361,85 @@ compactRoundTrip = forAll gen $ \v ->
 compactTest :: TestTree
 compactTest = testProperty "fromCompact . toCompact == id" compactRoundTrip
 
+coinCBORTests :: TestTree
+coinCBORTests =
+  testGroup
+    "Coin CBOR roundtrips"
+    [ testCase "decode 0 as Coin for pre-conway" $
+        forM_ [natVersion @2 .. natVersion @8] $ \version ->
+          roundTripVersionExpectation version (Coin (0))
+    , testCase "fail decode 0 as Coin for conway" $
+        roundTripFailureExpectation (natVersion @9) (Coin (0))
+    , testCase "decode -1 as Coin for pre-conway" $
+        forM_ [natVersion @2 .. natVersion @8] $ \version ->
+          roundTripVersionExpectation version (Coin (-1))
+    , testCase "fail decode -1 as Coin for conway" $
+        roundTripFailureExpectation (natVersion @9) (Coin (-1))
+    , testCase "decode 1 as Coin for all eras" $
+        roundTripCborExpectation (Coin (1))
```

Comment:
All these test currently pass and I think they are correct.

Reply:
Generate using `Positive` instead of `1`